tests/integration/helpers.py

```diff
@@ -57,3 +86,88 @@ async def deploy_charm(ops_test: OpsTest, charm: Path) -> None:
+        port: Port number of the URL.
+        protocol: Transfer protocol (default: http).
+
+    Returns:
+        Unit URL of the form {protocol}://{address}:{port}
+    """
+    status = await ops_test.model.get_status()
+    address = status["applications"][application]["units"][f"{application}/{unit}"]["address"]
+    return f"{protocol}://{address}:{port}"
+
+
+async def ensure_model(
+    label: str, ops_test: OpsTest, cloud_name: str, cloud_type: Literal["k8s", "lxd"], enable_debug_logs: bool = True
+) -> str:
+    """Get (or create) the model on the given cloud and return its name.
+
+    Args:
+        label: Unique label to ensure a distinct model across tests.
+        ops_test: PyTest object.
+        cloud_name: Name of the cloud from the Juju controller.
+        cloud_type: Type of the cloud between k8s and lxd.
+        enable_debug_logs: Whether to have debug logs enabled in the model.
+
+    Returns:
+        Name of the model being ensured.
+
+    Raises:
+        SetupFailedError: If model creation or configuration fails.
+    """
+    # Ref: https://github.com/canonical/livepatch-k8s-operator/blob/b48995d02f549dca0be46dd15790b5985669d7fe/tests/integration/helpers.py#L32  # noqa: W505
+    model_name = f"datahub-test-{label}-{cloud_name}-{__RUN_ID}"
+    # Juju limits model names to 63 characters. This does not guarantee name uniqueness!
+    model_name = model_name[:63]
+    if any((v.model_name == model_name for v in ops_test.models.values())):
+        return model_name
+
+    # Explicitly create the model to avoid auth errors.
+    code, stdout, stderr = await ops_test.juju("add-model", model_name, cloud_name)
+    if code != 0:
+        logger.error("'juju add-model' failed:\n\tstdout: %s\n\tstderr: %s", stdout, stderr)
+        raise exceptions.SetupFailedError("command 'juju add-model' failed")
+
+    model = await ops_test.track_model(
+        alias=model_name,
+        model_name=model_name,
+        cloud_name=cloud_name,
+        use_existing=True,
+        keep=True,  # TODO (mertalpt): Switched for debug purposed, rollback before merge.
+    )
+
+    # No need for further processing.
+    if cloud_type != "k8s":
+        return model.name
+
+    # There is a bug that prevents deploying charms for k8s cloud on lxd controller.
+    # Prevent by updating the workload storage parameter.
+    # See:
+    # - https://bugs.launchpad.net/juju/+bug/2031216
+    # - https://bugs.launchpad.net/juju/+bug/2077426
+    controller_cloud = await (await model.get_controller()).cloud()
+    if controller_cloud.cloud.type_ == "lxd":
+        code, stdout, stderr = await ops_test.juju(
+            "model-config", "-m", model.name, "workload-storage=microk8s-hostpath"
+        )
+        if code != 0:
+            logger.error("'juju model-config' failed:\n\tstdout: %s\n\tstderr: %s", stdout, stderr)
+            raise exceptions.SetupFailedError("command 'juju model-config' failed")
+
+    if enable_debug_logs:
```

Comment:
do you need logging enabled only when cloud_type is `k8s`?

Reply:
K8s is where DataHub is, so unless we hit something particularly nasty to debug this is what we need.